auth.go

```diff
@@ -0,0 +1,127 @@
+package mysql
+
+import "bytes"
+
+const mysqlClearPassword = "mysql_clear_password"
```

Comment:
Use
```go
const (
     mysqlClearPassword = "mysql_clear_password"
     mysqlNativePassword = "mysql_native_password"
     ...
)
```
for a list of constants like this. I think go vet or go fmt should also catch this.